pulpcore/tests/functional/__init__.py

```diff
@@ -117,6 +117,14 @@ def orphans_cleanup_api_client(pulpcore_client):
     return OrphansCleanupApi(pulpcore_client)
 
 
+@pytest.fixture
+def gen_role(roles_api_client, gen_object_with_cleanup):
```

Comment:
@mdellweg is this the right place to introduce this fixture?

Reply:
wasn't sure if they should go in `__init__.py` or `conftest.py`. I put it here because that's where all the other fixtures are. LMK if I should move it.

Reply:
After thinking more on it, I think this is the right place because the `roles_api_client` is offered from pulpcore itself. @mdellweg lmk if you think differently.

@newswangerd I'm trying to get our factory pattern usage to have `_factory` in it so can this be 
```suggestion
def role_factory(roles_api_client, gen_object_with_cleanup):
```